lib/pbench/test/unit/server/conftest.py

```diff
@@ -736,31 +700,131 @@ def create_drb_user(client, server_config, fake_email_validator):
 
 
 @pytest.fixture()
-def pbench_token(client, server_config, create_drb_user):
-    # Login user to get valid pbench token
-    response = login_user(client, server_config, "drb", generic_password)
-    assert response.status_code == HTTPStatus.OK
-    data = response.json
-    assert data["auth_token"]
-    return data["auth_token"]
+def pbench_admin_token(client, create_admin_user):
+    return generate_token(
+        user=create_admin_user,
+        username=admin_username,
+        pbench_client_roles=["ADMIN"],
+    )
+
+
+@pytest.fixture()
+def pbench_token(client, create_drb_user):
+    """
+    OIDC valid token for the 'drb' user.
+    """
+    return generate_token(username="drb", user=create_drb_user)
+
+
+@pytest.fixture()
+def pbench_token_invalid(client, create_drb_user):
+    """
+    OIDC invalid token for the 'drb' user
+    """
+    return generate_token(username="drb", user=create_drb_user, valid=False)
+
+
+def generate_token(
+    username: str,
+    user: Optional[User] = None,
+    pbench_client_roles: Optional[list[str]] = None,
+    valid: Optional[bool] = True,
+) -> str:
+    """
+    Generates an OIDC JWT token that mimics the real OIDC token obtained
+    from the OIDC complaint client login.
+
+    Args:
+        username: username to include in the token payload
+        user: user attributes will be extracted from the user to include in
+            the token payload, if not provided it will be queried from the User
+            table. It is a responsibility of the consumer to make sure the
+            user exists in the 'User' table until we remove the User table
+            completely.
+        pbench_client_roles: List of any roles to add in the token payload
+        valid: If True, the generated token will be valid for 10 mins.
+               If False, generated token would be invalid and expired
+
+    TODO: When we remove the User table we need to update this functionality's
+        dependance on user table.
+        Refer related Jira issues:
+            https://issues.redhat.com/browse/PBENCH-934,
+            https://issues.redhat.com/browse/PBENCH-900,
+            https://issues.redhat.com/browse/PBENCH-895,
+            https://issues.redhat.com/browse/PBENCH-962
+
+    Returns
+        JWT token string
+    """
+    # Current time to encode in the token payload
+    current_utc = datetime.datetime.now(datetime.timezone.utc)
+
+    # Expired epoch time to use as a token expiry in case the caller
+    # wants an invalid token.
+    EXPIRY = "1667187770"
+
+    if not user:
+        user = User.query(username=username)
+    if valid:
+        exp = current_utc + datetime.timedelta(minutes=10)
+    else:
+        exp = EXPIRY
```

Comment:
This looks like a nice spot for a ternary:
```
    exp = current_utc + datetime.timedelta(minutes=10) if valid else EXPIRY
```
FWIW, my love for the ternary operator was cemented when I was working on the the Intel Itanium processor whose architecture included the notion of a "predicate register".  This enabled the code to perform conditional operations, particularly assignments, without using a branch instruction.  (That is, it could issue the load instruction and decide later whether it should have any effect.)  And, the easiest way to prompt the compiler to produce a conditional load was with a ternary operator.

However, it's also handy from a source code perspective.  For instance, using a ternary means that there is exactly one assignment to the variable (or one function call), and it is the value which ends up being conditional, which means that the execution can never "get around" the assignment by taking another branch (or there's no question about which call site the function was invoked at).  And, modulo Black's assistance, it often results in more compact code (which is a Good Thing™, provided the result isn't too obscure).

Reply:
> FWIW, my love for the ternary operator was cemented when I was working on the the Intel Itanium processor whose architecture included the notion of a "predicate register". This enabled the code to perform conditional operations, particularly assignments, without using a branch instruction.

Wow; one doesn't often these days see positive references to IA-64. And, yeah; the instruction set was insane. (Try writing a thread context switch: I did.) But of course the relationship to ternary logic depends entirely on a smart compiler (though, to be sure, that was an inherent assumption & dependency of IA-64)... but, more importantly, I wonder what the chances are that *Python* was that smart about code emulation?

And given the blackness of our Python, ternaries so easily get unreadably ugly that I tend to automatically avoid them even when they'd fit. (I also really despise Python's ternary syntax, but that's a separate issue.)

Reply:
> Wow; one doesn't often these days see positive references to IA-64.

😆

> And, yeah; the instruction set was insane.

Um, yeah, as evidenced by the thickness of the architecture books...which, for some reason, are still here on my desk....

> Try writing a thread context switch: I did.

I was spared from that, but not from instruction-level debugging (I think I was looking at d-cache misses...).

> But of course the relationship to ternary logic depends entirely on a smart compiler (though, to be sure, that was an inherent assumption & dependency of IA-64)...

Ha!  Of course we depended on the compiler!  No human could write IA-64 assembly!  (Well...not very much, and not without suffering greatly for it...I no longer possess the recollection of how Java produced IA-64 code....  😉)

> more importantly, I wonder what the chances are that _Python_ was that smart about code emulation?

Um...I'm confused about your sense of importance....  🤣

> And given the blackness of our Python, ternaries so easily get unreadably ugly that I tend to automatically avoid them even when they'd fit. (I also really despise Python's ternary syntax, but that's a separate issue.)

I'm with you, except that I find that I cannot abandon them entirely...so, if the expression (with the indentation) will fit in 80 columns, I'll propose it.  😁

Reply:
> > And, yeah; the instruction set was insane.
> 
> Um, yeah, as evidenced by the thickness of the architecture books...which, for some reason, are still here on my desk....

I did my IA-64 working in the keycard access restricted raised-floor IA-64 lab during the initial port of Digital UNIX and OpenVMS, when the architecture references were individually registered with Intel by owner name and there were guards to shoot you if you tried to leave the room with them. (Well, not quite, but they were tracked carefully, and I couldn't say anything about what I did to the rest of the team. Our sources were in a separate repo on a secured server inside the lab.)

By the time we gave up on IA-64 Digital UNIX and got to leave the lab, the books were no longer top secret and I kept them. However I decided there are limits to that sort of sentimentality and got rid of them when I moved. It wasn't perhaps quite as bad as the full OpenVMS manual set, but it was awfully heavy and bulky!